envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactory.kt

```diff
@@ -24,24 +24,50 @@ class LuaFilterFactory(private val incomingPermissionsProperties: IncomingPermis
         null
     }
 
+    private val trustedClientIdentityHeader = incomingPermissionsProperties.trustedClientIdentityHeader
+
+    private val ingressScriptsMetadata = Metadata.newBuilder()
+        .putFilterMetadata("envoy.filters.http.lua",
+            Struct.newBuilder()
+                .putFields("client_identity_headers",
+                    Value.newBuilder()
+                        .setListValue(ListValue.newBuilder()
+                            .addAllValues(
+                                incomingPermissionsProperties.clientIdentityHeaders
+                                    .map { Value.newBuilder().setStringValue(it).build() }
+                            )
+                            .build()
+                        ).build()
+                )
+                .putFields("trusted_client_identity_header",
+                    Value.newBuilder()
+                        .setStringValue(trustedClientIdentityHeader)
+                        .build()
+                )
+                .putFields("san_uri_lua_pattern",
+                    Value.newBuilder()
+                        .setStringValue(
+                            SanUriMatcherFactory(incomingPermissionsProperties.tlsAuthentication)
```

Comment:
It would be better, but to achieve single-instance I would have to pass `SanUriMatcherFactory` through many, many classes. I don't know if it's worth it, because the instances of `SanUriMatcherFactory` are created only one time, at EC startup.

Reply:
I guess it's okay for now but I'm more worried about maintainability than performance. I think we should reconsider the composition/dependencies strategy in the project. To me `I would have to pass SanUriMatcherFactory through many, many classes` looks like a design issue.